lib/le/host/connection.rb

```diff
@@ -150,45 +142,36 @@ def write(message)
       end
 
       def start_async_thread
-        @thread = Thread.new { run() }
-        dbg "LE: Asynchronous socket writer started"
+        @thread = Thread.new { run }
+        dbg 'LE: Asynchronous socket writer started'
         @started = true
       end
 
       def check_async_thread
-        if not(@thread && @thread.alive?)
-          @thread = Thread.new { run() }
-        end
+        @thread = Thread.new { run } unless @thread&.alive?
```

Comment:
`thread&.alive?`? What does this magic syntax do?

Reply:
scroll down halfway, basically `if thread not nil and thread.alive`
https://www.rubyguides.com/2018/10/defined-keyword/